### PR TITLE
Added coverage test 0521

### PR DESCRIPTION
**Pull request by Vihanakangas**

### CODE REVIEW

File: coverage/PENDING_SUBMIT/amdvlk/0521-COVERAGE-tiny-ptr-vector-h-85/cov-nested-loops-float-bits-to-int-increment-array.amber [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+#         for(int j = 0; j < 5; j ++)
+#         {
+#             int idx = clamp(a, 0, 8);
+#             arr[idx] = _int_0;
+#
+#             for(int k = _int_0; k < i; k ++)
+#             {
+#                 for(int l = 0; l < j; l ++)
+#                 {
+#                     switch(a - _int_1065353216)
```
Comment (afd) [inline coverage/PENDING_SUBMIT/amdvlk/0521-COVERAGE-tiny-ptr-vector-h-85/cov-nested-loops-float-bits-to-int-increment-array.amber:70]: Add a comment to spell out the fact that a - _int_blah will be zero, which is not one of the switch cases, and thus the switch statement doesn't do anything.

**Review by afd — CHANGES_REQUESTED** [review]

File: coverage/PENDING_SUBMIT/amdvlk/0521-COVERAGE-tiny-ptr-vector-h-85/cov-nested-loops-float-bits-to-int-increment-array.amber (hunk 1 above)
Comment (asuonpaa) [reply coverage/PENDING_SUBMIT/amdvlk/0521-COVERAGE-tiny-ptr-vector-h-85/cov-nested-loops-float-bits-to-int-increment-array.amber:70]: a gets incremented later and all the cases do get executed.

Comment (afd) [reply coverage/PENDING_SUBMIT/amdvlk/0521-COVERAGE-tiny-ptr-vector-h-85/cov-nested-loops-float-bits-to-int-increment-array.amber:70]: I see - can you add some more commentary to explain what happens to arr[idx]? I think the point is that idx is guaranteed to be 8, so only arr[8] changes. But the current comment says that only arr[8] changes to zero, which made me think that the switch cases didn't have any effect. Can you make it more explicit when arr[8] changes and what value it ends up having?

**Review by afd — CHANGES_REQUESTED** [review]

**Comment by Vihanakangas** [comment]
Updated.

**Review by afd — APPROVED** [review]
Thanks - this is much clearer.

**Pull request merged**